Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,458 @@
+        new gd.PropertyDescriptor(objectContent.fontFamily)
+          .setType('string')
+          .setLabel(_('Base font family'))
+      );
+
+      objectProperties.set(
+        'visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        fontSize: '20',
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Single array approach to add all Setters/Getters, so we don't have to deal with SO much boilerplate repetition
+    // Could be useful for other extensions... Thinking of making an online tool to generate this file from json data some day
+    // 0 function name, 1 description, 2 icon name, 3 param property type, 4 hasModifier
+    [
+      [
+        'BBText',
+        'BBCode formatted text',
+        'textAlign24',
+        'string',
+        'BBCode text',
+      ],
+      ['Color', 'base color', 'color24', 'color', 'Color (Hex)'],
+      ['Opacity', 'base opacity', 'opacity24', 'number', 'Opacity (0-255)'],
+      ['FontSize', 'base font size', 'characterSize24', 'number', 'Font size'],
+      ['FontFamily', 'base font family', 'font24', 'string', 'Font family'],
+      [
+        'Alignment',
+        'text alignment',
+        'textAlign24',
+        'stringWithSelector',
+        'Alignment',
+        `["left", "right", "center"]`,
+      ],
+      [
+        'WrappingWidth',
+        'wrapping width',
+        'scaleWidth24',
+        'number',
+        'Wrapping width',
+      ],
+    ].forEach(p => {
```

Comment:
Not a big fan for using super short names like "p". Make it hard to convey what this "thing" even is :)
Prefer a descriptive name, like... "property"? It's a term quite generic and already used everywhere.. so we might thing of a better name. But at least it's a matter of search/replace to rename it, you can't do that with `p` :)